lib/absinthe/type/custom.ex

```diff
@@ -0,0 +1,96 @@
+defmodule Absinthe.Type.Custom do
+  use Absinthe.Schema.Notation
+
+  @moduledoc """
+  This module contains additional data types.
+
+  To use: `import_types Absinthe.Type.Custom`.
+  """
+
+  scalar :datetime, name: "DateTime" do
+    description """
+    The `DateTime` scalar type represents a date and time in the UTC
+    timezone. The DateTime appears in a JSON response as an ISO8601 formatted
+    string, including UTC timezone ("Z").
+    """
+
+    serialize &DateTime.to_iso8601/1
+    parse &parse_datetime/1
+  end
+
+  scalar :naive_datetime, name: "NaiveDateTime" do
+    description """
+    The `Naive DateTime` scalar type represents a naive date and time without
+    timezone. The DateTime appears in a JSON response as an ISO8601 formatted
+    string.
+    """
+
+    serialize &NaiveDateTime.to_iso8601/1
+    parse &parse_naive_datetime/1
+  end
+
+  scalar :date do
+    description """
+    The `Date` scalar type represents a date. The Date appears in a JSON
+    response as an ISO8601 formatted string.
+    """
+
+    serialize &Date.to_iso8601/1
+    parse &parse_date/1
+  end
+
+  scalar :time do
+    description """
+    The `Time` scalar type represents a time. The Time appears in a JSON
+    response as an ISO8601 formatted string.
+    """
+
+    serialize &Time.to_iso8601/1
+    parse &parse_time/1
+  end
+
+  @spec parse_datetime(any) :: {:ok, DateTime.t} | :error
+  defp parse_datetime(value) when is_binary(value) do
```

Comment:
These functions are not passed simply a string IE "2016-01-01T12:12:12". Rather they are passed an input blueprint struct. You'll want to pattern match that a string input has been given by doing:
```
defp parse_datetime(%Absinthe.Blueprint.Input.String{value: value}) do
```

When you do so, there isn't really any need to do the `is_binary` check.